tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c

```diff
@@ -815,33 +836,51 @@ static void test_other_group(void)
 	PASS();
 }
 
-#define TEST_CASE(role, name, description)                                     \
-	{                                                                      \
-		.test_id = "friendship_" #role "_" #name,                      \
-		.test_descr = description,                                     \
-		.test_post_init_f = test_##role##_init,                        \
-		.test_tick_f = bt_mesh_test_timeout,                           \
-		.test_main_f = test_##role##_##name,                           \
+/** LPN disable test.
+ *
+ * Check that toggling lpn_set() results in correct disabled state
+ */
+static void test_lpn_disable(void)
+{
+	bt_mesh_test_setup();
+
+	bt_mesh_lpn_set(true);
+	bt_mesh_lpn_set(false);
+
+	if (!evt_wait(LPN_ESTABLISHED, K_SECONDS(30))) {
```

Comment:
By looking at the test description in `lpn_disable.sh` and the lpn code, I think, the test could be more strict if it checks that no `LPN_POLLED` event fired. When establishing a friendship, Friend poll message is sent after Friend Offer message is received and the poll callback is called in this case. Checking `LPN_POLL` event will guarantee the _immediate_ reaction on `bt_mesh_lpn_set(false);` call. This still may not check the introduced fix if the underlying layers' behavior gets changed  and `friend_req_sent` callback is called right in the `bt_mesh_ctl_send` call.